google/api_core/retry_async.py

```diff
@@ -196,15 +292,25 @@ def __init__(
         self._timeout = kwargs.get("deadline", timeout)
         self._deadline = self._timeout
         self._on_error = on_error
+        self._is_stream = is_stream
 
-    def __call__(self, func, on_error=None):
+    def __call__(
+        self,
+        func: Callable[
+            ..., Awaitable[_R] | AsyncIterable[_Y] | Awaitable[AsyncIterable[_Y]]
+        ],
+        on_error: Callable[[BaseException], Any] | None = None,
+    ) -> Callable[_P, Awaitable[_R | AsyncGenerator[_Y, None]]]:
```

Comment:
I'm starting to wonder whether the streaming retries should use an entirely different class:
1) this type definition is complicated, and hard to deal with in downstream code
2) We treat Retry objects as something that end-users can build on their own, then pass into gapic methods. It doesn't really make sense to pass a retry object with `is_stream=True` to a unary method (or the other way around really, even though that's the default behaviour now)

Let me know what you think

Reply:
I made the change, explained in a separate comment